network/address.go

```diff
@@ -67,7 +68,81 @@ func (a Address) NetworkAddress() string {
 		return ""
 	}
 	vals := strings.Split(string(a), typeAddressSep)
-	return vals[1]
+	validNetworkAddress := vals[1]
+
+	host, port, err := net.SplitHostPort(validNetworkAddress)
+	if err != nil {
+		return ""
+	}
+
+	// checking the validity of the IP address (both IPv4 or IPv6)
+	if net.ParseIP(host) != nil {
+		return validNetworkAddress
+	}
+
+	// if host is not a valid IP address (i.e. ParseIP returns nil),
+	// it must be a hostname (since at the beginning we check whether the address is valid
+	ipAddress, err := net.LookupHost(host)
+	if err != nil {
+		return ""
+	}
+
+	sort.Strings(ipAddress)
```

Comment:
I did it so that the returned IP address is always the same and testing would be easier (LookupHost returns the address in a random order). Otherwise, test should be change in such a way that the test case takes all the IP addresses associated to the hostname, and we could just see whether the IP address returned from NetworkAddress is included in those IP addresses of the test.